src/common/tracer.cc

```diff
@@ -44,9 +46,83 @@ jspan Tracer::add_span(opentracing::string_view span_name, jspan& parent_span) {
   return noop_tracer->StartSpan(span_name);
 }
 
+jspan Tracer::add_span(opentracing::string_view span_name, const opentracing::SpanContext& parent_ctx) {
+  if (is_enabled()) {
+    return open_tracer->StartSpan(span_name, { opentracing::ChildOf(&parent_ctx) });
```

Comment:
if the decoding fails or the attribute does not exist, then `parent_ctx` will be empty.
need to make sure that this is be coming a no-op here